docs/architecture.md

```diff
@@ -0,0 +1,50 @@
+# Application Architecture
+This is just a sketch
+
+## General
+* Authentication
+  * Authenticate
+  * Permissions
+  * Roles
+  * Resources
+* Router
+  * Validate request
+  * Authenticate & authorise request
+  * Determine which controller to call based on request
+  * Forms the response of the controller into a well-formed API request
+* Controllers
+  * All application logic, delegate to appropriate Services and Models
+* Models
+  * Business logic relating to DB entities
+* Services
+  * E-mail
+  * Frontline
+* Scheduled Jobs
+
+## Diagram
+![[application architecture]](./assets/architecture.png)
+
+## Directory Structure
```

Comment:
Weirdly this isn't displaying properly in markdown. Might want a new line after?